src/scripts/main.js

```diff
@@ -1,5 +1,41 @@
 'use strict';
 
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
 document.addEventListener('click', (e) => {
-  // write code here
+  const wallRect = wall.getBoundingClientRect();
+  const spiderRect = spider.getBoundingClientRect();
+  const x = e.clientX;
+  const y = e.clientY;
+
+  const wallField = e.target.closest('.wall');
+
+  if (!wallField) {
+    return;
+  }
+
+  let leftSpider = x - wallRect.left - spiderRect.width / 2 - wall.clientLeft;
+  let topSpider = y - wallRect.top - spiderRect.height / 2 - wall.clientTop;
```

Comment:
The calculation for `leftSpider` and `topSpider` is correct, but it might be clearer to separate the calculation of the center offset into its own variable for readability. This would make it easier to understand the intention behind subtracting half the spider's width and height.